internal/provider/models.go

```diff
@@ -268,12 +268,17 @@ type Role struct {
 	ResourceId   types.String `tfsdk:"resource_id"`
 }
 
-type RoleGrant struct {
+type UserRoleGrants struct {
 	ID     types.String `tfsdk:"id"`
 	UserId types.String `tfsdk:"user_id"`
 	Roles  []Role       `tfsdk:"roles"`
 }
 
+type UserRoleGrant struct {
+	UserID types.String `tfsdk:"user_id"`
```

Comment:
I experimented with this and it doesn't look like we need it.  I did eventually run into an issue with the testing framework where we needed an ID but this was avoided by migrating to the new framework.   See here:
https://developer.hashicorp.com/terraform/plugin/framework/acctests#no-id-found-in-attributes

Here is the PR where I migrated the framework.
https://github.com/cockroachdb/terraform-provider-cockroach/pull/185

Also, I mentioned to you early on that we were planning to add a unique ID for this but decided against it because it didn't fit into our current public api definitions at all.